examples/car_control_complementarity/car_control_complementarity.py

```diff
@@ -0,0 +1,132 @@
+# This is an example use of path complementarities to enforce not braking and accelerating
+# at the same time.
+
+import nosnoc
+import casadi as ca
+import numpy as np
+import matplotlib.pyplot as plt
+
+X0 = np.array([0, 0])
+X_TARGET = np.array([500, 0])
+
+def car_model():
+    q = ca.SX.sym('q')
+    v = ca.SX.sym('v')
+    x = ca.vertcat(q, v)
+
+    u = ca.SX.sym('u',2)
+    lbu = np.zeros((2,))
+    ubu = np.ones((2,))
+
+    k1 = 5
+    k2 = 3
+    kb = 5
+
+    j_a = 1
+    j_b = 1
+
+    A = np.array([
+        [0, 1],
+        [0, 0]
+    ])
+
+    B1 = np.array([
+        [0,0],
+        [k1,-kb]
+        ])
+    B2 = np.array([
+        [0,0],
+        [k2,-kb]
+        ])
+
+    f_1 = A@x + B1@u
+    f_2 = A@x + B2@u
+
+    F = [ca.horzcat(f_1, f_2)]
+
+    c = [v-15]
+    S = [np.array([[-1],[1]])]
+
+    g_terminal = x - X_TARGET
+
+    f_q = j_a*u[0]**2 + j_b*u[1]**2
+
+    g_path_comp = ca.horzcat(u[0], u[1])
+
+    model = nosnoc.NosnocModel(x=x, F=F, S=S, c=c, x0=X0, u=u)
+    ocp = nosnoc.NosnocOcp(lbu=lbu, ubu=ubu, f_q=f_q, g_terminal=g_terminal, g_path_comp=g_path_comp)
```

Comment:
I thought that removing the path comp here should give the same solution.
Somehow, it does find another solution that seems to fulfill the complementarity but is less optimal, higher cost value.

Reply:
That is very strange. It looks to me like I get the same result.

Reply:
I double checked after pulling.
Without the new complementarity, I get:
```
-------------------------------------------
sigma 		 compl_res 	 nlp_res 	 cost_val 	 CPU time 	 iter 	 status
1.0e+00 	 1.22e+02 	 5.78e-01	 1.42e+01 	 1.221857 	 150 	 Solved_To_Acceptable_Level
1.0e-01 	 1.21e+01 	 1.00e-01	 6.02e+00 	 3.853913 	 500 	 Maximum_Iterations_Exceeded
Warning: IPOPT exited with status Maximum_Iterations_Exceeded
1.0e-02 	 1.25e+00 	 1.00e-02	 6.07e+00 	 0.412568 	 22 	 Solve_Succeeded
1.0e-03 	 1.26e-01 	 1.00e-03	 5.84e+00 	 1.001033 	 55 	 Solve_Succeeded
1.0e-04 	 1.22e-02 	 1.00e-04	 5.84e+00 	 0.335566 	 18 	 Solve_Succeeded
1.0e-05 	 1.21e-03 	 1.00e-05	 5.84e+00 	 0.347576 	 19 	 Solve_Succeeded
1.0e-06 	 1.20e-04 	 1.00e-06	 5.84e+00 	 0.270447 	 15 	 Solve_Succeeded
```

with complementarity:
```
/home/oj/nosnoc_py/nosnoc/ocp.py:72: UserWarning: OCP: using path complementarities. Note that all expressions a, b need to be bound as 0 <= a, 0 <= b. This is not yet done automatically.
  warn("OCP: using path complementarities. Note that all expressions a, b need to be bound as 0 <= a, 0 <= b. This is not yet done automatically.")
-------------------------------------------
sigma 		 compl_res 	 nlp_res 	 cost_val 	 CPU time 	 iter 	 status
1.0e+00 	 1.12e+02 	 1.00e+00	 5.05e+00 	 2.270347 	 283 	 Solved_To_Acceptable_Level
1.0e-01 	 1.19e+01 	 1.00e-01	 5.17e+00 	 0.178105 	 16 	 Solve_Succeeded
1.0e-02 	 1.31e+00 	 1.00e-02	 5.18e+00 	 0.117990 	 12 	 Solve_Succeeded
1.0e-03 	 1.31e-01 	 1.00e-03	 5.19e+00 	 0.089867 	 11 	 Solve_Succeeded
1.0e-04 	 1.26e-02 	 1.00e-04	 5.19e+00 	 0.100883 	 11 	 Solve_Succeeded
1.0e-05 	 1.23e-03 	 1.00e-05	 5.19e+00 	 0.114858 	 13 	 Solve_Succeeded
1.0e-06 	 1.21e-04 	 9.97e-07	 5.19e+00 	 0.131870 	 13 	 Solve_Succeeded
```

Maybe something to check again later or turn into a test.